src/mbgl/renderer/renderer_impl.cpp

```diff
@@ -339,6 +339,10 @@ void Renderer::Impl::render(const UpdateParameters& updateParameters) {
         } else {
             std::sort(sortedTiles.begin(), sortedTiles.end(),
                       [](const auto& a, const auto& b) { return a.get().id < b.get().id; });
+            // Don't render non-symbol layers for tiles that we're only holding on to for symbol fading
+            sortedTiles.erase(std::remove_if(sortedTiles.begin(), sortedTiles.end(),
+                                             [](const auto& tile) { return tile.get().tile.holdForFade(); }),
+                              sortedTiles.end());
```

Comment:
I looked at this a little bit. I think it would look something like: add a `getFadingRenderTiles()` method to `RenderSource`, and then implement it in the various children of that class (any source that doesn't have fading tiles can use a default empty implementation). In the renderer, just concatenate `getFadingRenderTiles()` onto `getRenderTiles()` for symbol layers. Then in `TilePyramid`, we could either maintain a parallel set of "hold for fading" tiles, or we could just do the filtering in the accessors (since internally to `TilePyramid` you usually want to iterate over all of them at the same time).

I think it could be an improvement, but it wasn't such an obvious simplification that I really wanted to get it in. Unless you're concerned, I'll defer it to future refactorings.